src/app/RecordingMetadata/BulkEditLabels.tsx

```diff
@@ -0,0 +1,163 @@
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import * as React from 'react';
+import {
+  Button,
+  Card,
+  CardBody,
+  CardHeader,
+  CardHeaderMain,
+  Stack,
+  StackItem,
+  Text,
+  TextVariants,
+} from '@patternfly/react-core';
+import { ServiceContext } from '@app/Shared/Services/Services';
+import { useSubscriptions } from '@app/utils/useSubscriptions';
+import { ArchivedRecording } from '@app/Shared/Services/Api.service';
+import { includesLabel, parseLabels, RecordingLabel } from './RecordingLabel';
+import { first, forkJoin, Observable } from 'rxjs';
+import { EditableLabelCell } from '@app/RecordingMetadata/EditableLabelCell';
+
+export interface BulkEditLabelsProps {
+  isTargetRecording: boolean;
+  checkedIndices: number[];
+  recordings: ArchivedRecording[];
+  hideForm: () => void;
+}
+
+export const BulkEditLabels: React.FunctionComponent<BulkEditLabelsProps> = (props) => {
+  const context = React.useContext(ServiceContext);
+  const [commonLabels, setCommonLabels] = React.useState([] as RecordingLabel[]);
+  const [prevCommonLabels, setPrevCommonLabels] = React.useState([] as RecordingLabel[]);
+  const [editing, setEditing] = React.useState(false);
+  const addSubscription = useSubscriptions();
+
+  const handleUpdateLabelsForSelected = React.useCallback(() => {
+    const tasks: Observable<any>[] = [];
+    const toDelete = prevCommonLabels.filter((label) => !includesLabel(commonLabels, label));
+
+    props.recordings.forEach((r: ArchivedRecording, idx) => {
+      if (props.checkedIndices.includes(idx)) {
+        let updatedLabels = [...parseLabels(r.metadata.labels), ...commonLabels];
+        updatedLabels = updatedLabels.filter((label) => {
+          return !includesLabel(toDelete, label);
+        });
+
+        tasks.push(
+          props.isTargetRecording
+            ? context.api.postTargetRecordingMetadata(r.name, updatedLabels).pipe(first())
+            : context.api.postRecordingMetadata(r.name, updatedLabels).pipe(first())
+        );
+      }
+    });
+    addSubscription(forkJoin(tasks).subscribe(() => setEditing(!editing)));
+  }, [
+    props.recordings,
+    props.checkedIndices,
+    props.isTargetRecording,
+    commonLabels,
+    parseLabels,
+    editing,
+    setEditing,
+    context,
+    context.api,
+  ]);
+
+  const handleEditLabels = React.useCallback(() => {
+    setEditing(!editing);
```

Comment:
This should be `setEditing(editing => !editing)`

Reply:
I fixed 2 out of 3 `setEditing` calls. The `props.setEditing` call in `BulkEditLabels` link 94 only seems to compile with `props.setEditing(!props.editing)`. If I change it to `props.setEditing(editing => !editing)`, the error reads `Argument of type '(editing: any) => boolean' is not assignable to parameter of type 'boolean'`.

Reply:
I think you just need to update this:

```typescript
  setEditing: (editing: boolean) => void;
```

in the `BulkEditLabelsProps` to suit the new signature.